e2e/testcases/basic_test.go

```diff
@@ -0,0 +1,342 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"kpt.dev/configsync/e2e/nomostest"
+	"kpt.dev/configsync/e2e/nomostest/ntopts"
+	nomostesting "kpt.dev/configsync/e2e/nomostest/testing"
+	v1 "kpt.dev/configsync/pkg/api/configmanagement/v1"
+	"kpt.dev/configsync/pkg/api/configsync"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	yamlDir = "../testdata"
+)
+
+func TestNoDefaultFieldsInNamespaceConfig(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.Reconciliation2, ntopts.SkipMultiRepo)
+	expectedName := "hello-world"
+	expectedTime := metav1.Time{}
+
+	nt.T.Log("Add a deployment")
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/dir-namespace.yaml", yamlDir), "acme/namespaces/dir/namespace.yaml")
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/deployment-helloworld.yaml", yamlDir), "acme/namespaces/dir/deployment.yaml")
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Adding deployment")
+	nt.WaitForRepoSyncs()
+
+	nt.T.Log("Check that the deployment was created")
+	if err := nt.Validate("hello-world", "dir", &appsv1.Deployment{}); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Check that specified field name is present")
+	if err := nt.Validate("dir", "", &v1.NamespaceConfig{},
+		func(o client.Object) error {
+			raw := o.(*v1.NamespaceConfig).Spec.Resources[0].Versions[0].Objects[0].Raw
+			dep := &appsv1.Deployment{}
+			if err := json.Unmarshal(raw, &dep); err != nil {
+				return errors.Errorf("Failed to get raw JSON, %s", err)
+			}
+			actualName := dep.GetName()
+			if actualName != expectedName {
+				return errors.Errorf("Expected name: %s, got: %s", expectedName, actualName)
+			}
+			return nil
+		}); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Check that unspecified field creationTimestamp is not present")
+	if err := nt.Validate("dir", "", &v1.NamespaceConfig{},
+		func(o client.Object) error {
+			raw := o.(*v1.NamespaceConfig).Spec.Resources[0].Versions[0].Objects[0].Raw
+			dep := &appsv1.Deployment{}
+			if err := json.Unmarshal(raw, &dep); err != nil {
+				return errors.Errorf("Failed to get raw JSON, %s", err)
+			}
+			actualTime := dep.GetCreationTimestamp()
+			if actualTime != expectedTime {
+				return errors.Errorf("Expected time: %s, got: %s", expectedTime, actualTime)
+			}
```

Comment:
You can merge the validation into the first `nt.Validate`.

Reply:
both validation predicate have been merged

Reply:
You can merge them into the same predicate.

Reply:
ah sorry i misunderstood the original feedback. the validation has been merged to a single predicate